File: front-end/src/app/nv/invoice/invoice.service.js
(function() {

    angular.module('app.nv.invoice').service('invoiceService', invoiceService);

    function invoiceService(invoiceFactory) {

        return {
            getInvoices: getInvoices,
            getInvoice: getInvoice,
            saveInvoice: saveInvoice,
            deleteInvoice: deleteInvoice,
            indexInvoice: indexInvoice
        }

        function getInvoices() {
            return invoiceFactory.list({});
        }

        function getInvoice(id) {
            return invoiceFactory.get({id}, null);
        }

        function saveInvoice(invoice) {
            return invoiceFactory.save({}, invoice);
        }

        function deleteInvoice(id) {
            return invoiceFactory.delete({id}, null);
        }

        function indexInvoice(search) {
            return invoiceFactory.index({},search);
        }

    }

})();

File: api/src/main/java/com/api/service/InvoiceService.java
package com.api.service;

import com.api.dto.InvoiceDTO;
import com.api.dto.InvoiceIndexDTO;
import com.api.model.Invoice;

import java.util.List;

public interface InvoiceService {

    Invoice dtoToEntity(InvoiceDTO invoiceDTO);

    List<InvoiceDTO> findAll();

    Invoice get(Long id);

    Invoice save(InvoiceDTO invoiceDTO);

    //Invoice update(InvoiceDTO invoiceDTO);

    void delete(Long id);

    List<InvoiceDTO> index(InvoiceIndexDTO search);
}
